### ISSUE TITLE
waterway length limit Spin box buttons miss step definitions

### ISSUE THREAD

**Issue by hessenfarmer** [widelands/widelands#5666]
The spin box buttons of the waterway length limit spinbox are not correctly translated see screenshot.    
furthermore with CTRL pressed it jumps to max

![shot0000](https://user-images.githubusercontent.com/43039887/202928359-9f7b2cf3-d8e3-4dc9-aa31-018699f60fd1.png)

**Comment by tothxa** [comment]
I think CTRL makes all spinboxes jump to min/max. Oh, I see, there is no max defined for length limit, so it jumps to `MAX_INT`. But 20 is only a recommended limit, not a hard one. Should this be changed, or should we define a separate hard limit too?

edit: I will do this part in #5653 if there is a decision on the hard limit.

**Comment by hessenfarmer** [comment]
Spin box design allows for normal step and big step definition. we could define them for the Waterways to 1 and 5 this would make reasonable steps with a recommended max of 20. I am not in favor of limitting it to a hard value.  
Setting 1 and 5 can be done in #5653 
the translation issue is maybe a different PR.

**Comment by tothxa** [comment]
The translation works for me in both hungarian and German. Maybe you just need to rebuild your translations?

edit: or you used a commit between https://github.com/widelands/widelands/commit/10347661765d1092855d540c4b62cc309707a556 and its German translation?

**Comment by tothxa** [comment]
> Setting 1 and 5 can be done in #5653 the translation issue is maybe a different PR.

OK. But CTRL will still jump to min/max. :)

**Comment by frankystone** [comment]
The possibility to use CTRL should also be mentioned in the tool tip.

Translation works for me.

Using CTRL let the value jump to MAX, but the image used for the warning is not set then. I would be fine if the max value is 100, with the stepping hessenfarmer proposed. A length of 100 would be a really big length for a ferry, imho.

**Comment by hessenfarmer** [comment]
After some thinking I think we safely can set 50 as max, as this would mean basically no restriction on waterway length.

**Comment by hessenfarmer** [comment]
I will rebuild master and recheck for the translations issue

**Comment by hessenfarmer** [comment]
After a clean build there is no translation issue anymore. Sorry. 

But now I think we should do it in the branch already open to implement big step and limit.

@tothxa would this be doable?

**Comment by tothxa** [comment]
Yes, I'll do it:

 - add upper limit of 50
 - make it `kBig` with 5 as big step[^1]
 - try to fix setting the warning image correctly when ctrl is used[^2]

[^1]: I already started this yesterday, but there's a minor annoyance: as "disabled" is actually 1, starting from it goes disabled→6→11→16… Should I try to make it 5→10→15? I haven't checked yet what it would need, so I don't know whether it's possible without touching the spinbox code.
[^2]: Again, I haven't checked whether the problem is in `spinbox.cc`

**Comment by hessenfarmer** [comment]
Personally i wouldn't mind if it is 6-11 or 5-10 so you take the desision whether the more organised look of 5-10 is worth the pain.

**Comment by tothxa** [comment]
I checked the missing update of the warning when ctrl is used to set min/max value, and it is indeed a spinbox bug: Ideally `set_value()` should call `changed()` (otherwise I need to add it in 6 places where CTRL is considered), but IIRC it caused an infinite loop when I tried that in one of my first PRs. Meanwhile I've found another bug in spinbox, that makes it possible to make `sbi->value` wrap around when max is `INT_MAX` and big step is used. So spinbox should get some overhaul, but I think that's out of the scope of #5653.

The first 2 is done.
